planner/core/preprocess.go

```diff
@@ -935,6 +935,9 @@ func checkColumn(colDef *ast.ColumnDef) error {
 				return types.ErrTooBigDisplayWidth.GenWithStackByArgs(colDef.Name.Name.O, mysql.MaxFloatingTypeWidth)
 			}
 		}
+		if tp.Flen < tp.Decimal {
```

Comment:
Below code should move into upper else block. float type can only provide `Flen` and the `Decimal` is `-1`. So current code could break `f float(64)`.

Reply:
ok